### DISCUSSION TITLE
求助：在新建注释时为满足条件的注释添加特定标签或者其他操作

### DISCUSSION THREAD

**Discussion by xjtusjtu** [windingwind/zotero-actions-tags#142] [Q&A] [answered]
求助各位大佬，我如何使用action tags插件，在新建注释时为满足条件的注释添加特定标签或者其他操作？我现在的问题是代码逻辑基本清楚，但不知道如何获取当前新建的注释对象，然后对其进行一系列操作

![image](https://github.com/windingwind/zotero-actions-tags/assets/57665402/fef65579-dc24-40bd-8712-3a4eba9f786c)

**Comment by xjtusjtu** [comment]
以下代码在Zotero 开发者-Run javascript中成功运行，但是在插件中按照上面图片配置后，按下快捷键无法得到目标结果。恳请大佬指正！
（在插件的自定义脚本中首行添加了 `const Zotero = require("Zotero");` ）

```
const window = Zotero.getMainWindow();
var Zotero_Tabs = window.Zotero_Tabs;

if(Zotero_Tabs._selectedID!= 'zotero-pane'){
	var div=Zotero.Reader.getByTabID(Zotero_Tabs._selectedID)._iframeWindow.document.querySelector('#annotations>div.annotation.selected');
	if(!(div && div.getAttribute("data-sidebar-annotation-id"))){
		// 若没有选中批注
		return "没有选中批注";
	}
} else {
	return "非PDF Reader窗口";
}

// Get the current annotation
let annotationID = div.getAttribute("data-sidebar-annotation-id");
let pdfitem = Zotero.Items.get(Zotero.Reader.getByTabID(Zotero_Tabs._selectedID).itemID);
let annotation = Zotero.Items.getByLibraryAndKey(pdfitem.libraryID, annotationID.replace("annotation-", ""));

// Check if the annotation's color is "#a28ae5"
if (annotation.annotationColor == "#a28ae5") {
    // Check if the annotation's content contains only one word
    let content = annotation.annotationText;
    // return annotation;
    if (content.trim().split(' ').length === 1) {
        // Add the tag "s/newWord" to the annotation
        annotation.addTag("snewWord");
        // Save the changes to the annotation
        await annotation.saveTx();
        return annotation;
        return "添加标签成功";
    }
}
```

**Comment by xjtusjtu** [comment]
进一步提问的，如果触发事件设置为添加注释，代码又需要如何修改？

**Comment by MuiseDestiny** [comment]
你这个问题本质是：阅读pdf时候界面，快捷键不被触发。

**Reply by xjtusjtu** [reply]
嗯嗯，所以需要有个例如“新建注释”的事件是吗？P佬

**Reply by windingwind** [reply, marked as answer]
使用event=create item即可捕获新建注释数据。

**Comment by Muskteer98** [comment]
根据您的代码，我做了一个，多个标签颜色的选择。
存在以下问题：
（1）采用阅读状态下，进行标注的话，可以添加标签。如图1方式。但是采用菜单栏上高亮文本的方式，就无法添加标签。
（2）在中文文献中可以顺利添加，在英文文献中无法顺利添加标签
（3）请教一下，这个“使用event=create item即可捕获新建注释数据”，应该如何修改代码？
（4）在已经定义了颜色名称的情况下，如何获取style中颜色的名称，可以减少下面的check部分代码，让后续在修改颜色名称后无须修改代码？
![image](https://github.com/windingwind/zotero-actions-tags/assets/29146415/045cfb21-87d0-4ceb-9699-4acafa71f7f0)


`const window = Zotero.getMainWindow();
var Zotero_Tabs = window.Zotero_Tabs;

if(Zotero_Tabs._selectedID!= 'zotero-pane'){
	var div=Zotero.Reader.getByTabID(Zotero_Tabs._selectedID)._iframeWindow.document.querySelector('#annotations>div.annotation.selected');
	if(!(div && div.getAttribute("data-sidebar-annotation-id"))){
		// 若没有选中批注
		return "没有选中批注";
	}
} else {
	return "非PDF Reader窗口";
}

// Get the current annotation
let annotationID = div.getAttribute("data-sidebar-annotation-id");
let pdfitem = Zotero.Items.get(Zotero.Reader.getByTabID(Zotero_Tabs._selectedID).itemID);
let annotation = Zotero.Items.getByLibraryAndKey(pdfitem.libraryID, annotationID.replace("annotation-", ""));

// Check if the annotation's color is one of the specified colors
if (
  annotation.annotationColor == "#ff9900" ||
  annotation.annotationColor == "#ffcc00" ||
  annotation.annotationColor == "#666666" ||
  annotation.annotationColor == "#ff0033" ||
  annotation.annotationColor == "#ff9999" ||
  annotation.annotationColor == "#339933" ||
  annotation.annotationColor == "#6699cc" ||
  annotation.annotationColor == "#9966cc"
) {
  // Check if the annotation's content contains only one word
  let content = annotation.annotationText;
  if (content.trim().split(' ').length === 1) {
    // Add the appropriate tag based on the annotation's color
    if (annotation.annotationColor == "#ff9900") {
      annotation.addTag("#摘要");
    } else if (annotation.annotationColor == "#ffcc00") {
      annotation.addTag("#背景");
    } else if (annotation.annotationColor == "#666666") {
      annotation.addTag("#概念");
    } else if (annotation.annotationColor == "#ff0033") {
      annotation.addTag("#重点");
    } else if (annotation.annotationColor == "#ff9999") {
      annotation.addTag("#讨论");
    } else if (annotation.annotationColor == "#339933") {
      annotation.addTag("#公式");
    } else if (annotation.annotationColor == "#6699cc") {
      annotation.addTag("#实验");
    } else if (annotation.annotationColor == "#9966cc") {
      annotation.addTag("#英文");
    }
    // Save the changes to the annotation
    await annotation.saveTx();
    return "添加标签成功";
  }
}

return "未添加标签";
`
